code/game/atoms.dm

```diff
@@ -275,6 +288,14 @@
 		if(canSmoothWith[length(canSmoothWith)] > MAX_S_TURF) //If the last element is higher than the maximum turf-only value, then it must scan turf contents for smoothing targets.
 			smoothing_flags |= SMOOTH_OBJ
 		SET_BITFLAG_LIST(canSmoothWith)
+	/// Money calculations here
+	if(!max_demand)	// If the item isnt getting a max_demand then give it a random one
+		max_demand = (5 * rand(5, 12)) // Makes sure it increases in increments of 5 - 6 * 5 = 25, 12 * 5 = 60
+	if(!custom_price)
+		if(w_class)
+			custom_price = (5 * rand(1, w_class * 4))
+		else
+			custom_price = rand(10, 20)
```

Comment:
(See above) Make this its own proc. On items add the w_class behaviour, but have the standard to be fully random.

```
/atom/proc/generate_price()
    custom_price = rand(10, 20)

/obj/item/generate_price()
    custom_price = (5 * rand(1, w_class * 4))
```

Reply:
If you override a proc, then the atom one never runs. In the example I sent, anything that inherits from /obj/item will run the second proc and not the first. If you want to run both then you can call ..() which is the operator to call the parent proc (in this case the parent proc is generate_price() defined on /atom)